File: HotelDemo/src/app/models/bill-dto.ts
// src/app/models/bill-dto.ts
export class BillDTO {
  billId?: number;
  reservationId: number;
  stayDates: string;
  serviceAmounts: number;
  taxes: number;
  total: number;

  constructor(
    billId: number = 0, // Provide default values
    reservationId: number = 0,
    stayDates: string = '',
    serviceAmounts: number = 0,
    taxes: number = 0,
    total: number = 0
  ) {
    this.billId = billId;
    this.reservationId = reservationId;
    this.stayDates = stayDates;
    this.serviceAmounts = serviceAmounts;
    this.taxes = taxes;
    this.total = total;
  }
}

File: DemoTwo/DemoTwo/DTOs/BillDTO.cs
using System.ComponentModel.DataAnnotations;

namespace DemoTwo.DTOs
{
    public class BillDTO
    {
        public int BillId { get; set; }

        [Required]
        public int ReservationId { get; set; }

        [Required]
        [StringLength(255)]
        public string StayDates { get; set; }

        [Range(0, double.MaxValue, ErrorMessage = "Service amounts must be zero or greater.")]
        public decimal ServiceAmounts { get; set; }

        [Range(0, double.MaxValue, ErrorMessage = "Taxes must be zero or greater.")]
        public decimal Taxes { get; set; }
        public decimal Total { get; set; }
    }
}
